step2-05/README.md

```diff
@@ -12,78 +12,80 @@ f(data) => view
+Actions are messages that represent some event, such as a user's action or a network request. With the aid of *reducers*, they affect the overall state.
 
-## Store
+### Store
 
-This is a singleton state tree. The state tree is immutable and needs to be re-created at every action. This helps connected views to know when to update itself - just doing a simple reference comparison rather than a deep comparison.
+The store contains a singleton state tree. The state tree is immutable and needs to be re-created at every action. This helps connected views easily to know when to update by allowing them to do a simple reference comparison rather than a deep comparison. You can think of each state as a snapshot of the app at that point in time.
 
-## Reducers
+### Dispatcher
 
-These are simple stateless, pure functions that takes state + action message and returns a copy of state some modifications according to the action message type and payload.
+There is a single dispatcher. It simply informs the store of all the actions that need to be performed. Additional middleware (explained later) can be applied to the store, and the dispatcher's job is to dispatch the message through all the middleware layers.
 
-## Dispatcher
+### Reducers
 
-There is a single dispatcher. It simply informs of the store of all the actions that needs to be performed. Certain middleware can be applied to the store and the dispatcher's job is to dispatch the message through all the middleware layers.
+Redux uses **reducers** to manage state changes. This name comes from the "reducer" function passed to `Array.reduce()`.
 
-Redux is used inside many large and complex applications because of its clarity and its predictability. It is really easy to debug and is easily extensible via its middleware architecture. In this exercise, we'll explore the heart of how Redux modifies state.
+A Redux reducer is a simple stateless **pure function** (no side effects). Its only job is to change the state from one immutable snapshot to another. It takes state + an action message as input, makes a modified copy of the state based on the action message type and payload, and returns the new state. (Reducers should not modify the previous state.)
 
-Redux uses what is called a "reducer" to modify its state. It is called this because a "reducer" is what is used inside an `Array.reduce()`.
+**Mental Model**: Think of a reducer as part of the store. It should have no side effects and only define how data changes from one state to the next given action messages.
 
-A reducer is a **pure function** that receives some state and an action message as inputs and generates a copy of modified state as the output. Redux manages state changes mainly through reducers, and they are directly related to the UI, so for this exercise, we'll have to use jest tests to see the inner workings.
+### Advanced: Middleware
```

Comment:
added a separate section mentioning middleware, but I don't love how short it is and how there are some other middleware-related points in random places before this